infra/gcp/terraform/modules/oci-proxy/network.tf

```diff
@@ -14,6 +14,153 @@ See the License for the specific language governing permissions and
+  ip_address            = google_compute_global_address.ipv4.address
+  port_range            = "443"
+  load_balancing_scheme = "EXTERNAL_MANAGED"
+}
+
+resource "google_compute_global_forwarding_rule" "http_ipv6" {
+  project               = var.project_id
+  name                  = "${var.project_id}-ipv6-http-managed"
+  target                = google_compute_target_http_proxy.default.self_link
+  ip_address            = google_compute_global_address.ipv6.address
+  port_range            = "80"
+  load_balancing_scheme = "EXTERNAL_MANAGED"
+}
+
+resource "google_compute_global_forwarding_rule" "https_ipv6" {
+  project               = var.project_id
+  name                  = "${var.project_id}-ipv6-https-managed"
+  target                = google_compute_target_https_proxy.default.self_link
+  ip_address            = google_compute_global_address.ipv6.address
+  port_range            = "443"
+  load_balancing_scheme = "EXTERNAL_MANAGED"
+}
+
+# Redirect HTTP to HTTPS
+resource "google_compute_target_http_proxy" "default" {
+  project = var.project_id
+  name    = "${var.project_id}-http-default"
+  url_map = google_compute_url_map.https_redirect.self_link
+}
+
+resource "google_compute_url_map" "https_redirect" {
+  project = var.project_id
+  name    = "${var.project_id}-http-to-https-redirect"
+  default_url_redirect {
+    https_redirect         = true
+    redirect_response_code = "MOVED_PERMANENTLY_DEFAULT"
+    strip_query            = false
+  }
+}
+
+# Serve HTTPS
+resource "google_compute_target_https_proxy" "default" {
+  project = var.project_id
+  name    = "${var.project_id}-default-https-proxy"
+  url_map = google_compute_url_map.default.self_link
+
+  # TODO: pivot to managing cert outside of lb-http module
+  ssl_certificates = var.ssl_certificates
+}
+
+resource "google_compute_region_network_endpoint_group" "default" {
+  for_each = google_cloud_run_service.oci-proxy
+
+  provider              = google-beta
+  project               = var.project_id
+  name                  = "${var.project_id}-${each.key}-neg"
+  network_endpoint_type = "SERVERLESS"
+  region                = google_cloud_run_service.oci-proxy[each.key].location
+  cloud_run {
+    service = google_cloud_run_service.oci-proxy[each.key].name
+  }
+}
+
+
+resource "google_compute_url_map" "default" {
+  project         = var.project_id
+  name            = "${var.project_id}-default-url-map"
+  default_service = google_compute_backend_service.default.self_link
+}
+
+resource "google_compute_backend_service" "default" {
+  project = var.project_id
+  name    = "${var.project_id}-default-bes"
+
+  enable_cdn = false
+
+  dynamic "backend" {
+    for_each = google_compute_region_network_endpoint_group.default
+    content {
+      group = backend.value.id
+    }
+  }
+
+  log_config {
+    enable      = true
+    sample_rate = 1.0
+  }
+
+  # TODO: outlier detection not working with terraform module:
```

Comment:
this will block using this seriously for now, but it won't block doing some manual testing.